[rpc] enable test_backward_node_failure for process group agent

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31580 [rpc] enable test_backward_node_failure for process group agent**
* #31023 [rpc] catch exceptions in ProcessGroupAgent::enqueueSend and report them.

Follow up to #31023. As per the title, this PR enables these 2 tests that were previously
disabled for process group agent, since after https://github.com/pytorch/pytorch/pull/31023, we do report exceptions on node failures for PG agent. These tests should now pass, and the exception message is determined via the type of RPC Agent being tested.

Differential Revision: [D19219515](https://our.internmc.facebook.com/intern/diff/D19219515/)

rohan-varma committed:
[rpc] enable test_backward_node_failure for process group agent

As per the title, this PR enables these 2 tests that were previously
disabled for process group agent, since after
https://github.com/pytorch/pytorch/pull/31023, we do report exceptions on node
failures for PG agent.

Differential Revision: [D19219515](https://our.internmc.facebook.com/intern/diff/D19219515/)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit b6aecad8:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=b6aecad8b341ad56777e834307682c325434bdb2).

### :detective: 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4055267) pytorch_linux_backward_compatibility_check_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=78588481) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=78588481))
<details>
<summary>
<code>Dec 23 22:45:18 unknown type specifier:</code>
</summary>

```
Dec 23 22:45:06 Looking in links: https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html 
Dec 23 22:45:08 Collecting torch 
Dec 23 22:45:09   Downloading https://download.pytorch.org/whl/nightly/cpu/torch-1.4.0.dev20191223%2Bcpu-cp36-cp36m-linux_x86_64.whl (127.2MB) 
Dec 23 22:45:15 Installing collected packages: torch 
Dec 23 22:45:17 Successfully installed torch-1.4.0.dev20191223+cpu 
Dec 23 22:45:18 + python check_backward_compatibility.py --new-schemas new_schemas.txt 
Dec 23 22:45:18 Traceback (most recent call last): 
Dec 23 22:45:18   File "check_backward_compatibility.py", line 86, in <module> 
Dec 23 22:45:18     s = parse_schema(line.strip()) 
Dec 23 22:45:18 RuntimeError:  
Dec 23 22:45:18 unknown type specifier: 
Dec 23 22:45:18 _TorchScriptTesting_StackString::pop(__torch__.torch.classes._TorchScriptTesting_StackString _0) -> (str _0) 
Dec 23 22:45:18                                      ~~~~~~~~~ <--- HERE 
Dec 23 22:45:18 =================== sccache compilation log =================== 
Dec 23 22:45:18 + cleanup 
Dec 23 22:45:18 + retcode=1 
Dec 23 22:45:18 + set +x 
Dec 23 22:45:18 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Dec 23 22:45:18 Compile requests                 0 
Dec 23 22:45:18 Compile requests executed        0 
Dec 23 22:45:18 Cache hits                       0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

rohan-varma committed:
Update on "[rpc] enable test_backward_node_failure for process group agent"


Follow up to #31023. As per the title, this PR enables these 2 tests that were previously
disabled for process group agent, since after https://github.com/pytorch/pytorch/pull/31023, we do report exceptions on node failures for PG agent. These tests should now pass, and the exception message is determined via the type of RPC Agent being tested.

Differential Revision: [D19219515](https://our.internmc.facebook.com/intern/diff/D19219515/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] enable test_backward_node_failure for process group agent"


Follow up to #31023. As per the title, this PR enables these 2 tests that were previously
disabled for process group agent, since after https://github.com/pytorch/pytorch/pull/31023, we do report exceptions on node failures for PG agent. These tests should now pass, and the exception message is determined via the type of RPC Agent being tested.

Differential Revision: [D19219515](https://our.internmc.facebook.com/intern/diff/D19219515/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] enable test_backward_node_failure for process group agent"


Follow up to #31023. As per the title, this PR enables these 2 tests that were previously
disabled for process group agent, since after https://github.com/pytorch/pytorch/pull/31023, we do report exceptions on node failures for PG agent. These tests should now pass, and the exception message is determined via the type of RPC Agent being tested.

Differential Revision: [D19219515](https://our.internmc.facebook.com/intern/diff/D19219515/)

[ghstack-poisoned]
